test/metalearn/metafeatures/compare_with_openml.py

```diff
@@ -0,0 +1,126 @@
+import openml
+import json
+import os
+import arff
+import numpy as np
+import traceback
+import sys
+
+from test.data.dataset import _read_arff_dataset
+from metalearn import Metafeatures
+
+FILE_PATH = './test/metalearn/metafeatures/openmlComparisons/'
+
+def compare_with_openml(runs=1, max_features=200, max_instances=50000, verbose=False):
+    # get a list of datasets from openml
+    datasets = _list_dataset_ids(max_features, max_instances)
+
+    inconsistencies = False
+    successful_runs = 0
+    sample_size = runs
+    while successful_runs < sample_size:
+        dataset_id = np.random.choice(datasets, replace = False)
+        print(f"Dataset_id {dataset_id}: ", end="")
+        sys.stdout.flush()
+        try:
+            dataset = _download_dataset(dataset_id)
+            if dataset is None:
+                print(f"Invalid dataset type")
+            else:
+                if _compare_metafeatures(dataset, dataset_id, verbose):
+                    inconsistencies = True
+                successful_runs += 1
+                print(f"Succeeded. Runs = {successful_runs}")
+        except KeyboardInterrupt:
+            raise
+        except Exception as e:
+            print(f"Error: {e}")
+            _write_results(traceback.format_exc(), dataset_id)
+                
+    if inconsistencies:
+        print(f"\nNot all metafeatures matched results from OpenML.\nResults written to {FILE_PATH}")       
+        
+def _list_dataset_ids(max_features, max_instances):
+    datasets_dict = openml.datasets.list_datasets()
+    datasets = [
+        k for k,v in datasets_dict.items() if 
+        ("NumberOfInstances" in v.keys() and v["NumberOfInstances"] <= max_instances) and 
+        ("NumberOfFeatures" in v.keys() and v["NumberOfFeatures"] <= max_features)
+    ]
+    return datasets
+
+def _download_dataset(dataset_id):
+    raw_dataset = openml.datasets.get_dataset(dataset_id)
+    target = str(raw_dataset.default_target_attribute).split(",")
+    df = _read_arff_dataset(raw_dataset.data_file)
+    if len(target) <= 1:
+        if target[0] == "None":
+            X = df
+            Y = None
+        else:
+            X = df.drop(columns=target, axis=1)
+            Y = df[target].squeeze()
+        dataset_metafeatures = {x: (float(v) if v is not None else v) for x,v in raw_dataset.qualities.items()}
```

Comment:
Why cast to float? After a few runs, it seems to me that they are all already floats. Are there any counterexamples?

Reply:
I'm not really sure, It might be a workaround to some error either Seth or I got at some point, but errors can pretty hard to reproduce. It's definitely possible though that it's unneccessary